api/src/main/java/jakarta/servlet/http/HttpSession.java

```diff
@@ -213,4 +219,30 @@ public interface HttpSession {
      */
     public boolean isNew();
 
+    /**
+     * Provides a mechanism for applications to interact with the {@code HttpSession} outside of the scope of an HTTP
+     * request.
+     * <p>
+     * To use this mechanism, applications call the {@code getAccessor()} method on {@code HttpSession} instance that will be accessed.
+     * This call must be made from within the scope of an HTTP request and will return a  {@code Consumer<Consumer<HttpSession>>}
+     * that then can be used outside of the scope of a HTTP request to access the session.   Such access is obtained by passing a 
+     * {@code Consumer<HttpSession>>} to the {@code Consumer#accept(Object)} method on the {@code Consumer} instance
+     * returned from this method.  During that call, the container will call the the {@code Consumer#accept(Session)} method of the 
+     * application provided {@code Consumer<Session>} passing an {@code HttpSession} object that represents the same 
+     * {@code HttpSession} that the accessor was obtained from, but it may be a different instance and have been passivated then activated
+     * since the accessor was obtained.  If the session has been invalidated before the call, then the consumer will be called with an invalid 
+     * {@code HttpSession} instance representing the original {@code HttpSession}. 
+     
+     * <p>
+     * For the purposes of session access, validity, passivation, activation etc. the container behaves as if the call the
+     * container makes to the {@code Consumer#accept(Session)} method of the application provided {@code Consumer<Session>}
+     * instance occurs during the processing of an HTTP request.
+     *
+     * @return A container provided {@code Consumer} instance that accepts application provided {@code Consumer<Session>}
+     * instances to enable the application to interact with the HTTP session outside of the scope of an HTTP request or
+     * {@code null} if access to the session is not supported outside of an HTTP request
+     */
+    default public Consumer<Consumer<HttpSession>> getAccessor() {
+        return null;
+    }
```

Comment:
```suggestion
    /**
     * An accessor for applications to interact with a {@code HttpSession} outside of the scope of an HTTP request.
     */
    interface Accessor {
        /**
         * Call to access the {@code HttpSession} used to obtain this {@code Accessor} from outside the scope of a HTTP request.
         * <p>
         * When called, the container will call the {@link Consumer#accept(Object)} method of the {@code sessionConsumer}
         * passed by the application, with an {@code HttpSession} object that represents the same {@code HttpSession} that this
         * {@code Accessor} was obtained from.
         * <p>
         * The passed {@code HttpSession} may be the same instance as used to obtain this {@code Accessor}, or a new instance 
         * for the session which has possibly been passivated and activated since the {@code Accessor} was obtained.
         * <p>
         * The passed {@code HttpSession} may be shared concurrently with other {@code Accessor}s for the same session,
         * other calls to this {@code Accessor} and/or {@link jakarta.servlet.Servlet}s and {@link jakarta.servlet.Filter}s.
         * <p>
         * The passed {@code HttpSession} must not be used or referenced outside the scope of the call
         * to the {@link Consumer#accept(Object)} method of {@code sessionConsumer}.
         * <p>
         * For the purposes of session access, validity, passivation, activation etc. the container behaves as if the call 
         * to the {@link Consumer#accept(Object)} method of {@code sessionConsumer} occurs during the processing of an HTTP 
         * request for the same session.
         *
         * @param sessionConsumer the application provided {@link Consumer} of {@code HttpSession} to access the session.
         * @throws IllegalStateException if this method is called on an invalidated session or if the session ID has
         *                               changed since the {@code Accessor} was obtained.
         */
         void access(Consumer<HttpSession> sessionConsumer);
    }

    /**
     * Provides a mechanism for applications to interact with this session outside the scope of an HTTP request.
     *
     * @return A container provided {@link Accessor} for this session or {@code null} if access is not supported outside 
     *         an HTTP request.
     * @throws IllegalStateException if this method is called on an invalidated session
     */
    default Accessor getAccessor() {
        return null;
    }

```